File: php/tests/SAMLAuthTest.php
<?php

namespace Tests\Auth;

use Auth\SAMLAuth;

class SAMLAuthTest extends TestUtils
{
    private $auth;

    protected function setUp(): void
    {
        parent::setUp();
        $this->auth = new SAMLAuth(
            'test-entity-id',
            'test-assertion-consumer-service-url',
            'test-idp-sso-url',
            'test-idp-certificate'
        );
    }

    public function testValidSAMLResponse(): void
    {
        $request = $this->createTestRequest('POST', [
            'Content-Type' => 'application/x-www-form-urlencoded'
        ], 'SAMLResponse=test-saml-response');
        $this->runAuthTest($this->auth, $request, 200, false);
    }

    public function testInvalidSAMLResponse(): void
    {
        $request = $this->createTestRequest('POST', [
            'Content-Type' => 'application/x-www-form-urlencoded'
        ], 'SAMLResponse=invalid-saml-response');
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testMissingSAMLResponse(): void
    {
        $request = $this->createTestRequest('POST', [
            'Content-Type' => 'application/x-www-form-urlencoded'
        ], '');
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testInvalidContentType(): void
    {
        $request = $this->createTestRequest('POST', [
            'Content-Type' => 'application/json'
        ], '{}');
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testInvalidHTTPMethod(): void
    {
        $request = $this->createTestRequest('GET');
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testGetType(): void
    {
        $this->assertAuthType($this->auth, 'saml');
    }
} 

File: go/src/auth/saml_auth_test.go
package auth

import (
	"net/http"
	"testing"
)

func TestSAMLAuth(t *testing.T) {
	// Create a new SAMLAuth instance with test configuration
	auth := NewSAMLAuth(
		"test-entity-id",
		"test-assertion-consumer-service-url",
		"test-idp-sso-url",
		"test-idp-certificate",
	)

	// Test cases
	tests := []TestCase{
		{
			Name: "Valid SAML assertion",
			Request: CreateTestRequest("POST", map[string]string{
				"Content-Type": "application/x-www-form-urlencoded",
			}, "SAMLResponse=test-saml-response"),
			ExpectedStatus: http.StatusOK,
			ExpectedError:  false,
		},
		{
			Name: "Invalid SAML assertion",
			Request: CreateTestRequest("POST", map[string]string{
				"Content-Type": "application/x-www-form-urlencoded",
			}, "SAMLResponse=invalid-saml-response"),
			ExpectedStatus: http.StatusUnauthorized,
			ExpectedError:  true,
		},
		{
			Name: "Missing SAML response",
			Request: CreateTestRequest("POST", map[string]string{
				"Content-Type": "application/x-www-form-urlencoded",
			}, ""),
			ExpectedStatus: http.StatusUnauthorized,
			ExpectedError:  true,
		},
		{
			Name: "Invalid content type",
			Request: CreateTestRequest("POST", map[string]string{
				"Content-Type": "application/json",
			}, "{}"),
			ExpectedStatus: http.StatusUnauthorized,
			ExpectedError:  true,
		},
		{
			Name: "Invalid HTTP method",
			Request: CreateTestRequest("GET", nil),
			ExpectedStatus: http.StatusUnauthorized,
			ExpectedError:  true,
		},
	}

	for _, tc := range tests {
		t.Run(tc.Name, func(t *testing.T) {
			RunAuthTest(t, auth, tc)
		})
	}
}

func TestSAMLAuthGetType(t *testing.T) {
	auth := NewSAMLAuth("entity-id", "acs-url", "idp-url", "certificate")
	if auth.GetType() != "saml" {
		t.Errorf("GetType() = %v, want %v", auth.GetType(), "saml")
	}
} 